src/data_types/strs.rs

```diff
@@ -161,6 +163,45 @@ impl CStr16 {
             pos: 0,
         }
     }
+
+    /// Write a string slice into the provided buffer. If it fails, then most probably, because
+    /// the buffer is not big enough. In that case, the buffer will contain the correct string
+    /// until the point, where the size was not enough.
+    ///
+    /// ## Example
+    ///
+    /// ```rust
+    /// use uefi::data_types::ArrayString;
+    /// use uefi::{CStr16, Char16};
```

Comment:
Just a minor style suggestion: you can add a `#` before these lines to [hide them in the generated documentation](https://doc.rust-lang.org/rust-by-example/testing/doc_testing.html#motivation-behind-documentation-tests).